crates/fj-viewer/src/graphics/vertices.rs

```diff
@@ -47,19 +51,41 @@ impl Vertices {
 
     pub fn push_cross(
         &mut self,
-        position: Point<f64, 3>,
+        position: Point<3>,
         normal: [f32; 3],
         color: [f32; 4],
     ) {
         let d = 0.05;
 
         self.push_line(
-            [position - vector![d, 0., 0.], position + vector![d, 0., 0.]],
+            [
+                Point::from_array([
+                    position.x.into_f64() - d,
+                    position.y.into_f64(),
+                    position.z.into_f64(),
+                ]),
+                Point::from_array([
+                    position.x.into_f64() + d,
+                    position.y.into_f64(),
+                    position.z.into_f64(),
+                ]),
+            ],
             normal,
             color,
         );
         self.push_line(
-            [position - vector![0., d, 0.], position + vector![0., d, 0.]],
+            [
+                Point::from_array([
+                    position.x.into_f64(),
+                    position.y.into_f64() - d,
+                    position.z.into_f64(),
+                ]),
+                Point::from_array([
+                    position.x.into_f64(),
+                    position.y.into_f64() + d,
+                    position.z.into_f64(),
+                ]),
+            ],
```

Comment:
See comment above:
```suggestion
            [
                position - Vector::from([0., d, 0.]),
                position + Vector::from([0., d., 0.]),
                ]),
            ],
```

Reply:
Hmm, that sounds wrong. `Point` + `Vector` and `Point` - `Vector` should both result in a `Point`, as both operations basically move a point by an offset (the vector). I think you've discovered a bug!